code/controllers/subsystem/vote.dm

```diff
@@ -356,26 +338,12 @@ SUBSYSTEM_DEF(vote)
 		if("custom")
 			if(usr.client.holder)
 				initiate_vote("custom",usr.key)
-		else
-			submit_vote(round(text2num(href_list["vote"])))
-	usr.vote()
-
-/datum/controller/subsystem/vote/proc/remove_action_buttons()
-	for(var/v in generated_actions)
-		var/datum/action/vote/V = v
-		if(!QDELETED(V))
-			V.remove_from_client()
-			V.Remove(V.owner)
-	generated_actions = list()
-
-/mob/verb/vote()
-	set category = "OOC"
-	set name = "Vote"
+		if("vote")
+			submit_vote(round(text2num(params["index"])))
+	return TRUE
 
-	var/datum/browser/popup = new(src, "vote", "Voting Panel")
-	popup.set_window_options("can_close=0")
-	popup.set_content(SSvote.interface(client))
-	popup.open(FALSE)
+/datum/controller/subsystem/vote/ui_close(mob/user)
+	voting -= user.client
```

Comment:
remove the ckey from the list, since that is what you add

Reply:
Done